src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -2067,14 +2089,130 @@ export class StudyViewPageStore {
             this.initialVisibleAttributesClinicalDataBinCountData.isPending ||
             this.initialVisibleAttributesClinicalDataCountData.isPending ||
             this.mutationProfiles.isPending ||
-            this.cnaProfiles.isPending
+            this.cnaProfiles.isPending ||
+            this.fetchUserSettings.isPending
         ) {
             return true;
         } else {
             return false;
         }
     }
 
+    private updateSettingsTimeout: Timer;
+    private studySettings: any = {};
+
+    @autobind
+    updateUserSettings(chartVisibilityUpdated: boolean, chartTypeUpdated: boolean, customChartAdded: boolean) {
```

Comment:
I would rather separate this function to three su-functions. They are not inter connected and you have do all if check.

Reply:
it depends on how we model `studySettings`, whether these attributes goes this way or a normalized for each chart